digits/static/js/layer_visualizations/readCaffeModel.js

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2016, NVIDIA CORPORATION.  All rights reserved.
+function generateCaffeTree(text){
+  window.graph = {nodes: new Array(), links: new Array()};
+  window.couplings = new Array();
+
+  // Cleans up attributes loaded from parsing a prototxt file
+  function cleanAttr(attr,name){return attr.replace(name,'').replace(/["" '']/g,'');}
+
+  // Get attributes from an string blob loaded from a prototxt file
+  function getAttr(data,name){
+    var attr = _.filter(data,function(d){return d.indexOf(name) > -1});
+    if (attr.length > 1 && name == "bottom:"){addCouples(data,attr);}
+    var d = attr[0];
+    if (_.isUndefined(d)){return undefined}
+    return cleanAttr(d,name);
+  }
+
+  // Get the location of all layer objects in prototxt file
+  function getLayerIndeces(data){
+    return _.without(_.map(data,function(d,i){
+      return d.indexOf("layer {") > -1 || d.indexOf("layers {") > -1 ? i : null;
```

Comment:
Also, it looks like you support old-style network definitions, but how do you handle types specified with an enum?
https://github.com/BVLC/caffe/blob/rc/examples/mnist/lenet.prototxt#L7-L9